_ru/tour/basics.md

```diff
@@ -26,187 +26,306 @@ previous-page: tour-of-scala
 
+{% tabs named-function %}
+{% tab 'Scala 2 и 3' for=named-function %}
 ```scala mdoc
 val addOne = (x: Int) => x + 1
 println(addOne(1)) // 2
 ```
+{% endtab %}
+{% endtabs %}
 
 Функции могут принимать множество параметров.
 
+{% tabs multiple-parameters %}
+{% tab 'Scala 2 и 3' for=multiple-parameters %}
 ```scala mdoc
 val add = (x: Int, y: Int) => x + y
 println(add(1, 2)) // 3
 ```
+{% endtab %}
+{% endtabs %}
 
 Или вообще не принимать никаких параметров.
 
+{% tabs no-parameters %}
+{% tab 'Scala 2 и 3' for=no-parameters %}
 ```scala mdoc
 val getTheAnswer = () => 42
 println(getTheAnswer()) // 42
 ```
+{% endtab %}
+{% endtabs %}
 
 ## Методы
 
 Методы выглядят и ведут себя очень похоже на функции, но между ними есть несколько принципиальных различий.
 
 Методы задаются ключевым словом `def`.  За `def` следует имя, список параметров, возвращаемый тип и тело.
 
+{% tabs method %}
+{% tab 'Scala 2 и 3' for=method %}
 ```scala mdoc:nest
 def add(x: Int, y: Int): Int = x + y
 println(add(1, 2)) // 3
 ```
+{% endtab %}
+{% endtabs %}
 
 Обратите внимание, как объявлен возвращаемый тип сразу _после_ списка параметров и двоеточия `: Int`.
 
 Методы могут принимать несколько списков параметров.
 
+{% tabs multiple-parameter-lists %}
+{% tab 'Scala 2 и 3' for=multiple-parameter-lists %}
 ```scala mdoc
 def addThenMultiply(x: Int, y: Int)(multiplier: Int): Int = (x + y) * multiplier
 println(addThenMultiply(1, 2)(3)) // 9
 ```
+{% endtab %}
+{% endtabs %}
 
 Или вообще ни одного списка параметров.
 
+{% tabs no-parameter-lists %}
+{% tab 'Scala 2 и 3' for=no-parameter-lists %}
 ```scala mdoc
 def name: String = System.getProperty("user.name")
 println("Hello, " + name + "!")
 ```
+{% endtab %}
+{% endtabs %}
 
 Есть некоторые отличия от функций, но пока что их можно рассматривать как нечто похожее.
 
 Методы также могут иметь многострочные выражения.
 
+
+{% tabs get-square-string class=tabs-scala-version %}
+
+{% tab 'Scala 2' for=get-square-string %}
 ```scala mdoc
 def getSquareString(input: Double): String = {
   val square = input * input
   square.toString
 }
 println(getSquareString(2.5)) // 6.25
 ```
+{% endtab %}
+
+{% tab 'Scala 3' for=get-square-string %}
+```scala
+def getSquareString(input: Double): String =
+  val square = input * input
+  square.toString
+
+println(getSquareString(2.5)) // 6.25
+```
+{% endtab %}
+
+{% endtabs %}
 
 Последнее выражение в теле становится возвращаемым значением метода (у Scala есть ключевое слово `return`, но оно практически не используется).
 
 ## Классы
 
 Вы можете объявлять классы используя ключевое слово `class`, за которым следует его имя и параметры конструктора.
 
+{% tabs greeter-definition class=tabs-scala-version %}
+
+{% tab 'Scala 2' for=greeter-definition %}
 ```scala mdoc
 class Greeter(prefix: String, suffix: String) {
   def greet(name: String): Unit =
     println(prefix + name + suffix)
 }
 ```
+{% endtab %}
+
+{% tab 'Scala 3' for=greeter-definition %}
+```scala
```

Comment:
Mb here `mdoc` and below could be added also as well?

```suggestion
```scala mdoc
```

Reply:
`"Tour of Scala"` compiles under 2.13. :(